perceval/backends/core/pagure.py

```diff
@@ -0,0 +1,404 @@
+        before raising a RetryError exception
+    :param max_items: max number of category items (e.g., issues,
+        pull requests) per query
+    :param archive: collect issues already retrieved from an archive
+    :param from_archive: it tells whether to write/read the archive
+    :param ssl_verify: enable/disable SSL verification
+    """
+    EXTRA_STATUS_FORCELIST = [403, 500, 502, 503]
+
+    def __init__(self, namespace, repository, token,
+                 sleep_time=DEFAULT_SLEEP_TIME, max_retries=MAX_RETRIES,
+                 max_items=MAX_CATEGORY_ITEMS_PER_PAGE, archive=None, from_archive=False, ssl_verify=True):
+        self.namespace = namespace
+        self.repository = repository
+        self.token = token
+        self.max_items = max_items
+
+        # URL to fetch the data from
+        base_url = PAGURE_API_URL
+
+        super().__init__(base_url, sleep_time=sleep_time, max_retries=max_retries,
+                         extra_headers=self._set_extra_headers(),
+                         extra_status_forcelist=self.EXTRA_STATUS_FORCELIST,
+                         archive=archive, from_archive=from_archive, ssl_verify=ssl_verify)
+
+    def issues(self, from_date=None):
+        """Fetch the issues from the repository.
+
+        The method retrieves, from a Pagure repository, the issues
+        updated since the given date.
+
+        :param from_date: obtain issues updated since this date
+
+        :returns: a generator of issues
+        """
+        payload = {
+            'status': 'all',
+            'per_page': self.max_items,
+            'order': 'asc'
+        }
+
+        if from_date:
+            payload['since'] = from_date
+
+        path = urijoin("issues")
+        return self.fetch_items(path, payload)
+
+    def fetch(self, url, payload=None, headers=None):
+        """Fetch the data from a given URL.
+
+        :param url: link to the resource
+        :param payload: payload of the request
+        :param headers: headers of the request
+        :param auth: auth of the request
+
+        :returns a response object
+        """
+        try:
+            response = super().fetch(url, payload, headers)
+        except requests.exceptions.HTTPError as error:
+            if error.response.status_code == 404 and error.response.reason == 'NOT FOUND':
+                logger.warning("The issue tracker is disabled please enable the feature for the repository")
+                return None
+            else:
+                raise error
+
+        return response
+
+    def fetch_items(self, path, payload):
+        """Return the items from Pagure API using links pagination
+
+        :param path: Path from which the item is to be fetched
+        :param payload: Payload to be added to the request
+
+        :returns: an item object
+        """
+        page = 0  # current page
+        last_page = None  # last page
+        if self.namespace:  # if project is under a namespace
```

Comment:
I would move the if-else block to a private method. A possible implementation could be https://github.com/chaoss/grimoirelab-perceval/blob/master/perceval/backends/core/launchpad.py#L326.

The first param of the urijoin should be self.base_url